compiler/test/stdlib/bytes.test.gr

```diff
@@ -0,0 +1,191 @@
+import Bytes from "bytes"
+import Int32 from "int32"
+import Array from "array"
+
+// Bytes.empty
+assert Bytes.empty == Bytes.empty
+
+
+// Bytes.length
+assert Bytes.length(Bytes.empty) == 0
+
+
+// Bytes.make
+let bytes = Bytes.make(64)
+assert Bytes.length(bytes) == 64
+
+
+// Bytes.setInt8, Bytes.getInt8S, Bytes.getInt8U
+let bytes = Bytes.make(1)
+Bytes.setInt8(0, 0x000000ffl, bytes)
+assert Bytes.getInt8S(0, bytes) == 0xffffffffl
+assert Bytes.getInt8U(0, bytes) == 0x000000ffl
+
+
+// Bytes.setInt16, Bytes.getInt16S, Bytes.getInt16U
+let bytes = Bytes.make(8)
+Bytes.setInt16(0, 0x0000ffffl, bytes)
+Bytes.setInt16(2, 0x00000001l, bytes)
+Bytes.setInt16(4, 0x00000002l, bytes)
+Bytes.setInt16(6, 0xffffff9cl, bytes)
+assert Bytes.getInt16S(0, bytes) == 0xffffffffl
+assert Bytes.getInt16U(0, bytes) == 0x0000ffffl
+assert Bytes.getInt16S(1, bytes) == 0x000001ffl
+assert Bytes.getInt16U(1, bytes) == 0x000001ffl
+assert Bytes.getInt16S(2, bytes) == 0x00000001l
+assert Bytes.getInt16U(2, bytes) == 0x00000001l
+assert Bytes.getInt16S(3, bytes) == 0x00000200l
+assert Bytes.getInt16U(3, bytes) == 0x00000200l
+assert Bytes.getInt16S(4, bytes) == 0x00000002l
+assert Bytes.getInt16U(4, bytes) == 0x00000002l
+assert Bytes.getInt16S(5, bytes) == 0xffff9c00l
+assert Bytes.getInt16U(5, bytes) == 0x00009c00l
+assert Bytes.getInt16S(6, bytes) == 0xffffff9cl
+assert Bytes.getInt16U(6, bytes) == 0x0000ff9cl
+
+
+// Bytes.setInt32, Bytes.getInt32
+let bytes = Bytes.make(4)
+Bytes.setInt32(0, 0x7fffffffl, bytes)
+assert Bytes.getInt32(0, bytes) == 0x7fffffffl
+
+
+// Bytes.setFloat32, Bytes.getFloat32
+let bytes = Bytes.make(4)
+Bytes.setFloat32(0, 42.0f, bytes)
+assert Bytes.getFloat32(0, bytes) == 42.0f
+
+
+// Bytes.setInt64, Bytes.getInt64
+let bytes = Bytes.make(8)
+Bytes.setInt64(0, 0x7fffffffffffffffL, bytes)
+assert Bytes.getInt64(0, bytes) == 0x7fffffffffffffffL
+
+
+// Bytes.setFloat64, Bytes.getFloat64
+let bytes = Bytes.make(8)
+Bytes.setFloat64(0, 42.0d, bytes)
+assert Bytes.getFloat64(0, bytes) == 42.0d
+
+
+// Bytes.copy
+let bytes0 = Bytes.make(8)
+assert Bytes.length(bytes0) == 8
+for (let mut i = 0; i < 8; i+= 1) {
+  Bytes.setInt8(i, Int32.fromNumber(i), bytes0)
+}
```

Comment:
I really want a `Bytes.init` for this pattern, maybe the syntax would be:
```
Bytes.init(8, (bytes0, i) => {
  Bytes.setInt8(i, Int32.fromNumber(i), bytes0)
})
```
@ospencer what do you think?

Reply:
We didn't go for a `Bytes.init` because we decided against adding a `Byte` type to Grain, at least for now. We could do it with `Int32`s that represent bytes, but that doesn't feel quite right.

Reply:
🤷